posthog/utils.py

```diff
@@ -365,3 +366,52 @@ def get_machine_id() -> str:
     # MAC addresses are 6 bits long, so overflow shouldn't happen
     # hashing here as we don't care about the actual address, just it being rather consistent
     return hashlib.md5(uuid.getnode().to_bytes(6, "little")).hexdigest()
+
+
+def get_table_size(table_name):
+    from django.db import connection
+
+    query = (
+        f'SELECT pg_size_pretty(pg_total_relation_size(relid)) AS "size" '
+        f"FROM pg_catalog.pg_statio_user_tables "
+        f"WHERE relname = '{table_name}'"
+    )
+    cursor = connection.cursor()
+    cursor.execute(query)
+    return dict_from_cursor_fetchall(cursor)
+
+
+def get_table_approx_count(table_name):
+    from django.db import connection
+
+    query = f"SELECT reltuples::BIGINT as \"approx_count\" FROM pg_class WHERE relname = '{table_name}'"
+    cursor = connection.cursor()
+    cursor.execute(query)
+    return dict_from_cursor_fetchall(cursor)
+
+
+def is_postgres_alive() -> bool:
+    from posthog.models import User
+
+    try:
+        User.objects.count()
+        return True
+    except DatabaseError:
+        return False
+
+
+def is_redis_alive() -> bool:
+    try:
+        return get_redis_heartbeat() != "offline"
+    except BaseException:
+        return False
+
+
+def get_redis_info() -> dict:
+    redis_instance = redis.from_url(settings.REDIS_URL, db=0)
```

Comment:
While trying this on a fresh Heroku instance I got the following error occasionally: https://sentry.io/share/issue/1a569d8da5b543198bbc05a129f759b5/

Likely because I was refreshing _system_status/ too much. Worth catching this error and displaying it in the frontend instead of dropping the entire request.